streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImplTest.java

```diff
@@ -404,6 +405,31 @@ public void shouldNotRemoveOffsetsOfUnUpdatedTablesDuringCheckpoint() throws Exc
         assertThat(updatedCheckpoint.get(t1), equalTo(101L));
     }
 
+    @Test
+    public void shouldSkipNullKeysWhenRestoring() throws Exception {
+        final HashMap<TopicPartition, Long> startOffsets = new HashMap<>();
+        startOffsets.put(t1, 1L);
+        final HashMap<TopicPartition, Long> endOffsets = new HashMap<>();
+        endOffsets.put(t1, 2L);
+        consumer.updatePartitions(t1.topic(), Collections.singletonList(new PartitionInfo(t1.topic(), t1.partition(), null, null, null)));
+        consumer.assign(Collections.singletonList(t1));
+        consumer.updateEndOffsets(endOffsets);
+        consumer.updateBeginningOffsets(startOffsets);
+        consumer.addRecord(new ConsumerRecord<>(t1.topic(), t1.partition(), 1, (byte[]) null, "null".getBytes()));
+        final byte[] expectedKey = "key".getBytes();
+        final byte[] expectedValue = "value".getBytes();
+        consumer.addRecord(new ConsumerRecord<>(t1.topic(), t1.partition(), 2, expectedKey, expectedValue));
+
+        stateManager.initialize(context);
+        final TheStateRestoreCallback stateRestoreCallback = new TheStateRestoreCallback();
+        stateManager.register(store1, false, stateRestoreCallback);
+        assertEquals(1, stateRestoreCallback.restored.size());
```

Comment:
because we have to compare `KeyValue<byte[], byte[]>` and `assertThat()` won't work., i.e., 2 `byte[]` with the same contents aren't equal. Need to use `assertArrayEquals()`